File: Learn/DataStructure/数据结构与算法之美/src/ConsoleApp1/ConsoleApp1/ArrayQueue.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ConsoleApp1
{
    class ArrayQueue<T>
    {
        private T[] items;
        private int n = 0;
        private int head = 0;
        private int tail = 0;

        public ArrayQueue(int capacity)
        {
            items = new T[capacity];
            n = capacity;
        }

        public bool Enqueue(T item)
        {
            if (tail == n) return false;
            items[tail] = item;
            ++tail;
            return true;
        }

        public T Dequeue()
        {
            if (head == tail) return default;
            T t = items[head];
            ++head;
            return t;
        }
    }
}

File: Learn/DataStructure/数据结构与算法之美/src/Clion/ArrayQueue.cpp
#include "ArrayQueue.h"
#include <stdio.h>

SqlList createList(int length){
    struct SqlList l;

    for (int i = 0; i < length; i++) {
        l.data[i] = i;
    }
    l.length = length;
    return l;
}

bool enqueue(String item){
    // 如果tail == n 表示队列已经满了
    if (tail) return false;
}

string dequeue(){
    if (head == tail) return NULL;

}